src/App.tsx

```diff
@@ -15,47 +18,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export function getSortedGoods(
+  goods: string[],
+  state: State,
```

Comment:
you should not pass the state inside the function. We should use only those props we need.
What's more, I believe that reverse logic should be separated. 

The point is the function ideally has but one single responsibility, it should not do a lot of stuff.